src/EditorFeatures/Core/NavigationBar/NavigationBarController_ModelComputation.cs

```diff
@@ -43,6 +85,12 @@ internal partial class NavigationBarController
 
         var model = await ComputeModelAsync().ConfigureAwait(false);
 
+        // If we were computing with frozen partial semantics here, enqueue work to compute *without* frozen
+        // partial snapshots so we move to accurate results shortly. Create and pass along a new cancellation
+        // token for this expensive work so that it can be canceled by future lightweight work.
+        if (frozenPartialSemantics)
+            _computeModelQueue.AddWork(new NavigationBarQueueItem(FrozenPartialSemantics: false, _nonFrozenComputationCancellationSeries.CreateNext(default)));
```

Comment:
It seems like this should just live in the caller -- you already have the if/else if, and then that way there is only one function doing queue management, and one function doing actual data computation.

Reply:
sgtm!

Reply:
Thanks.  that was cleaner!